src/Util/Help.php

```diff
@@ -0,0 +1,626 @@
+
+        foreach ($filteredOptions as $category => $options) {
+            // Initial state set to "true" to prevent a spacer at the start of an array.
+            $lastWasSpacer = true;
+            $spacerCount   = 0;
+
+            foreach ($options as $name => $option) {
+                if ($lastWasSpacer !== true && strpos($name, 'blank-line') === 0) {
+                    ++$spacerCount;
+                    $lastWasSpacer = true;
+                    continue;
+                }
+
+                if (in_array($name, $this->requestedOptions, true) === false) {
+                    unset($filteredOptions[$category][$name]);
+                    continue;
+                }
+
+                $lastWasSpacer = false;
+            }
+
+            // Make sure the final array doesn't contain a spacer at the end.
+            if (empty($filteredOptions[$category]) === false) {
+                end($filteredOptions[$category]);
+                $key = key($filteredOptions[$category]);
+                if (strpos($key, 'blank-line') === 0) {
+                    unset($filteredOptions[$category][$key]);
+                    --$spacerCount;
+                }
+            }
+
+            // Remove categories now left empty.
+            if (empty($filteredOptions[$category]) === true || count($filteredOptions[$category]) === $spacerCount) {
+                unset($filteredOptions[$category]);
+            }
+        }//end foreach
+
+        $this->activeOptions = $filteredOptions;
+
+    }//end filterOptions()
+
+
+    /**
+     * Determine the length of the longest option argument and store it.
+     *
+     * @return void
+     */
+    private function setMaxOptionNameLength()
+    {
+        $lengths = [];
+        foreach ($this->activeOptions as $category => $options) {
+            foreach ($options as $option) {
+                if (isset($option['argument']) === false) {
+                    continue;
+                }
+
+                $lengths[] = strlen($option['argument']);
+            }
+        }
+
+        if (empty($lengths) === false) {
+            $this->maxOptionNameLength = max($lengths);
+        }
+
+    }//end setMaxOptionNameLength()
+
+
+    /**
+     * Get the maximum width which can be used to display the help info.
+     *
+     * Independently of user preference/auto-determined width of the current screen,
+     * a minimum width is needed to display information, so don't allow this to get too low.
+     *
+     * @return int
+     */
+    private function getMaxWidth()
+    {
+        return max(self::MIN_WIDTH, $this->config->reportWidth);
+
+    }//end getMaxWidth()
+
+
+    /**
+     * Get the maximum width for the text in the option description column.
+     *
+     * @return int
+     */
+    private function getDescrColumnWidth()
+    {
+        return ($this->getMaxWidth() - $this->maxOptionNameLength - $this->indentWidth - $this->gutterWidth);
+
+    }//end getDescrColumnWidth()
```

Comment:
The method below this uses the full word `Description` in its name. I think it makes sense to change `Descr` to `Description` for this method as well. 🙂

Reply:
That's a fair point. I've updated the PR and implemented your suggestion.